woocommerce/woocommerce
# [Experimental] Product filters/active block new structure

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/51641

This PR refactors the active filter to the new design structure. Here are some additional notes.

- Fixed clear button setting where toggling it was not working in the editor/frontend.
- Changed the design of the chips to the new spec.
- Remove list view setting as we're just wanting to only chips.
- I did not implement the editor active chips (real data) in this PR. I think we can address this separately because one idea is we could actually use the same data from Attributes filter. So instead of calling the API multiple times, we call it once in the parent Product Filters block and pass the data through context to the child filters.
- Note that I found a bug where the clear button has to be above the filter list/chips block. This bug exists with other filters blocks as well and not specific to this block. But it doesn't block this PR. We can fix that bug in another PR. See step 7 to get around this bug and test this PR.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

> [!NOTE]  
> **Pre-requisite**
> - If you're using WC Beta Tester plugin, make sure the `experimental-blocks` feature is enabled (Tools > WCA Test Helper > Features).
> - Make sure to reset Product Catalog Template.

1. Navigate to Product Catalog template (Appearance > Editor > Templates > Product Catalog template.
2. Edit the template.
3. With the block inserter, add Product Filters (Experimental) block to the template.
4. Open the Document Overview (List view) and ensure you see the Product Filter Active block in the new design structure.
5. Select the parent Active Filters block, check the inspector controls and ensure you see the Clear button setting there.
6. Ensure you can enable/disable the Clear button and have it reflected in the editor.
7. Move the clear button to above the active chips block and save. View in the frontend and ensure the Clear button reflects your saved changes by choosing some filters.
8. Back in the editor Document Overview, ensure you see the Chips block. You should also see placeholder active chips in the editor.
9. Select the Chips block and in the inspector controls you should see option to change colors of the text, border and background. Have a play around with those and ensure they reflect in the editor and in the frontend.
10. In the frontend shop page (assuming you added the filter to the shop page), have a play around with the attributes and ensure the Active filter is working correctly by displaying the selected attributes.
11. Try removing a single active selected attribute by clicking on the X (remove button) on the chip. Ensure it removes the attribute filter and from the URL.
12. Now select more than one attributes and ensure you see the Clear button next to the active filter title.
13. Click on the Clear button and ensure it clears all the filters and in the URL.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
